### PR TITLE
Adding ASTQuery

### PR DESCRIPTION
**Pull request by Alcolo47**
ObjTest alone

### CODE REVIEW

**Comment by boxed** [comment]
Is this trying to solve the same problem as the ASTPattern class in mutmut? https://github.com/boxed/mutmut/blob/master/tests/test_mutation.py#L29 i.e. matching complex patterns on the AST.

It would be interesting to write the patterns we want to match using both systems and comparing how they read if so.

**Comment by abingham** [comment]
This looks mostly fine to me. The only change I'd like is to put the code into its own module. `ObjTest` is closely associated with the AST, so I propose this:

1. Make `ast.py` into a package. `ast/__init__.py` can expose all of the same stuff it does today.
2. Put the `ObjTest` implementation into `ast/obj_test.py`.

With that, I think everything would be good.

**Comment by Alcolo47** [comment]
OK, I will move this class. But about its name, I was not inspired, maybe AstTester ?

**Comment by abingham** [comment]
> maybe AstTester?

How about ASTQuery? It seems like it's primarily for querying for subsets of an AST.

File: src/cosmic_ray/operators/util.py [hunk 1]
```diff
@@ -19,3 +19,134 @@ def dec(cls):
...
+            return self
+        return self._clone(self.obj(*args, **kwargs))
+
+    def __getitem__(self, item) -> 'ObjTest':
+        if self.obj is None:
+            return self
+        return self._clone(self.obj[item])
+
+
+class ObjTestOptionnal(ObjTest):
```
Comment (abingham) [inline src/cosmic_ray/operators/util.py]: Do you mean `ObjTestOptional`? There's only on 'n' in 'optional'.

**Comment by abingham** [comment]
If you can fix the breaking tests, then I can merge this in.

**Pull request merged**